src/pages/ShoppingListPage.re

```diff
@@ -35,19 +37,41 @@ let reducer = (state, action) => {
       isLoading: false,
       shoppingList: list,
     }
-  | ApiCallError(errorMessage) => {
-      ...state,
-      isLoading: false,
-      error: Some(errorMessage),
-    }
+  | ApiCallError(errorMessage) =>
+    Js.log(errorMessage);
+    {...state, isLoading: false, error: Some(errorMessage)};
   | AddShoppingItem =>
-    let newItem: shoppingItem = {product: "", note: "", assignee: None};
+    let newItem: shoppingItem = {
+      product: "",
+      note: "",
+      assignee: None,
+      isPurchased: false,
+    };
     let items = Array.append(state.shoppingList.items, [|newItem|]);
     updateItems(state, items);
   | ChangeShoppingItem(index, newItem) =>
-    let updateFn = _item => newItem;
-    let items = updateWithIndex(state.shoppingList.items, updateFn, index);
-    updateItems(state, items);
+    if (state.shoppingList.status != Editing) {
+      state;
+    } else {
+      let updateFn = _item => newItem;
```

Comment:
Why `_item` and not just `item`? Is it some kind of a convention?

Reply:
Exactly!